ci/prow/make_config.go

```diff
@@ -699,6 +774,48 @@ func strExists(arr []string, str string) bool {
 	return false
 }
 
+// executeJobTemplateWrapper takes in consideration of repo settings, decides how many varianats of the
+// same job needs to be generated and generates them.
+func executeJobTemplateWrapper(repoName string, data interface{}, generateOneJob func(data interface{})) {
+	var legacyBranches []string
+	// Find out if LegacyBranches is set in repo settings
+	for _, repo := range repositories {
+		if repo.Name == repoName {
+			if len(repo.LegacyBranches) > 0 {
+				legacyBranches = repo.LegacyBranches
+			}
+		}
+	}
+	if len(legacyBranches) == 0 { // Generate only one job as normal if LegacyBranches is not set
+		generateOneJob(data)
+	} else {
+		// Generate one job with 'knative.dev' path alias for branches other than legacy branches,
+		// and another job without path alias for legacy branches
+		var base *baseProwJobTemplateData
+		switch v := data.(type) {
+		case *presubmitJobTemplateData:
+			base = &data.(*presubmitJobTemplateData).Base
+		case *postsubmitJobTemplateData:
+			base = &data.(*postsubmitJobTemplateData).Base
+		default:
+			log.Fatalf("unrecognized type: '%v'", v)
+		}
+		branches := base.Branches
+		ignoredBranches := base.SkipBranches
+		base.PathAlias = ""
+		base.Branches, base.SkipBranches = consolidateBranches(base.Branches, base.SkipBranches, legacyBranches, make([]string, 0))
+		generateOneJob(data)
+		base.Branches = branches
```

Comment:
If you generate this job first, you don't have to save `branches` or `ignoredBranches`, no?

Reply:
actually still need to save them, since `branches` and `skipbranches` both need to be computed based off of original `branches` and `skipBranches`. I have made the code show this clearer